src/IPHlpApi.Tests/IPHlpApiFacts.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright © .NET Foundation and Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net.NetworkInformation;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+using PInvoke;
+using Xunit;
+
+public class IPHlpApiFacts
+{
+    [Fact]
+    public void GetExtendedTcpTableTest()
+    {
+        IntPtr tcpTable = IntPtr.Zero;
+        int tcpTableLength = 0;
+
+        if (IPHlpApi.GetExtendedTcpTable(tcpTable, ref tcpTableLength, sort: true, AddressFamily.InterNetwork, IPHlpApi.TcpTableType.OwnerPidAll, 0) == Win32ErrorCode.ERROR_INSUFFICIENT_BUFFER)
+        {
+            try
+            {
+                tcpTable = Marshal.AllocHGlobal(tcpTableLength);
```

Comment:
Tip: using pointers in C# instead of these `IntPtr` types actually result in less code, and is often simpler to verify correctness.